packages/transport/src/modules/tokens/transmission/TokenContentRelationshipTemplate.ts

```diff
@@ -22,7 +25,19 @@ export class TokenContentRelationshipTemplate extends Serializable implements IT
     @serialize()
     public forIdentity?: CoreAddress;
 
+    @validate({ nullable: true, regExp: /^(pw|pin(4|5|6|7|8|9|10|11|12|13|14|15|16))$/ })
+    @serialize()
+    public passwordType?: string;
+
+    @validate({ nullable: true, customValidator: (v: ICoreBuffer) => (v.buffer.byteLength === 16 ? undefined : "must be 16 bytes long") })
+    @serialize()
+    public salt?: CoreBuffer;
+
     public static from(value: ITokenContentRelationshipTemplate): TokenContentRelationshipTemplate {
-        return this.fromAny(value);
+        const content = this.fromAny(value);
+        if (!content.passwordType !== !content.salt) {
+            throw TransportCoreErrors.general.invalidTruncatedReference("It's not possible to have only one of passwordType and salt set..");
```

Comment:
```suggestion
            throw TransportCoreErrors.general.invalidTruncatedReference("It's not possible to have only one of passwordType and salt set.");
```
you don't have to be sad about wrong usage